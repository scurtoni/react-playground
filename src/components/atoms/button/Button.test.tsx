import React from "react";
import {render, screen} from "@testing-library/react";
import renderer from "react-test-renderer";

import Button from "./Button";

test("render active button", () => {
  const button = <Button label="button" />;
  //snapshot test
  const tree = renderer.create(button).toJSON();
  expect(tree).toMatchSnapshot();
  //component test
  render(button);
  expect(screen.getByRole("button")).toBeInTheDocument();
});

test("render disabled button", () => {
  const button = <Button label="disabled button" disabled={true} />;
  //snapshot test
  const tree = renderer.create(button).toJSON();
  expect(tree).toMatchSnapshot();
  //component test
  render(button);
  expect(screen.getByRole("button")).toBeDisabled();
});
