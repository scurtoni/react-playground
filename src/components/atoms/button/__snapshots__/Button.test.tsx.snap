// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render active button 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium storybook-button storybook-button--medium storybook-button--secondary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "backgroundColor": undefined,
    }
  }
  tabIndex={0}
  type="button"
/>
`;

exports[`render disabled button 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium storybook-button storybook-button--medium storybook-button--secondary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
  disabled={true}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    {
      "backgroundColor": undefined,
    }
  }
  tabIndex={-1}
  type="button"
/>
`;
